client/CPlayerInterface.cpp

```diff
@@ -912,7 +920,13 @@ void CPlayerInterface::battleEnd(const BattleResult *br)
 
 		if(!battleInt)
 		{
-			GH.pushIntT<CBattleResultWindow>(*br, *this);
+			bool replay = !settings["adventure"]["alwaysSkipCombat"].Bool(); //do not allow manual replay

```

Comment:
Are you sure we do not need to test something else as well here? If we press autobattle in regular fight?

Reply:
Could be different corner cases which are not tested. For single player it's tested, autofight from UI as well